sonar-python-plugin/src/main/java/org/sonar/plugins/python/GeneratedIPythonFile.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python;
+
+import java.io.IOException;
+import java.util.Map;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.plugins.python.IpynbNotebookParser.IPythonLocation;
+
+public class GeneratedIPythonFile implements PythonInputFile {
+
+  InputFile wrappedFile;
+
+  Map<Integer, IPythonLocation> offsetMap;
+
+  private String pythonContent;
+
+  public GeneratedIPythonFile(InputFile wrappedFile, String pythonContent, Map<Integer, IPythonLocation> offsetMap) {
```

Comment:
I think adding the location map and offset map should be done in the ticket that needs it, that way we avoid going back and forth for something that can still change.